rq/worker.py

```diff
@@ -262,6 +261,24 @@ def __init__(self, queues, name: t.Optional[str] = None, default_result_ttl=DEFA
         elif exception_handlers is not None:
             self.push_exc_handler(exception_handlers)
 
+    def _set_connection(self, connection: Optional['Redis'], default_worker_ttl: int) -> 'Redis':
+        """Configures the Redis connection to have a socket timeout.
+        This should timouet the connection in case any specific command hangs at any given time (eg. BLPOP).
+        If the connection provided already has a `socket_timeout` defined, skips.
+
+        Args:
+            connection (Optional[Redis]): The Redis Connection.
+            default_worker_ttl (int): The Default Worker TTL
+        """
+        if connection is None:
+            connection = get_current_connection()
+        current_socket_timeout = connection.connection_pool.connection_kwargs.get("socket_timeout")
+        if current_socket_timeout is None:
+            timeout = max(1, default_worker_ttl - 15) + 20
```

Comment:
Nitpick: 10 seconds higher is enough.

Also, can we move the timeout calculation to a function so we can put the timeout calculation logic in one place?

Reply:
This comment also has not been addressed :)

Reply:
Yeah, still need to fix this, since this logic is used when setting the connection (before this attribute is set on the class). Will get back to this in a couple of days.

Reply:
@ronlut here